src/jit/gentree.cpp

```diff
@@ -15099,25 +15392,24 @@ void Compiler::gtCheckQuirkAddrExposedLclVar(GenTreePtr tree, GenTreeStack* pare
 }
 
 //------------------------------------------------------------------------
-// gtCanOptimizeTypeEquality: check if tree is a suitable input for a type
+// gtCanOptimizeTypeEquality: check if tree is a suitable input for type
```

Comment:
You may want to rename this method as "Can" typically implies that it returns a bool. Maybe gtOptimizableTypeEqualityKind() instead?

Reply:
How about `gtGetTypeProducerKind` ?